back/boxtribute_server/graph_ql/resolvers.py

```diff
@@ -219,26 +219,33 @@ def resolve_qr_code(obj, _, qr_code=None):
 
 
 @box.field("tags")
-def resolve_box_tags(box_obj, _):
-    return (
-        Tag.select()
-        .join(TagsRelation)
-        .where(
-            (TagsRelation.object_id == box_obj.id)
-            & (TagsRelation.object_type == TaggableObjectType.Box)
-        )
-    )
+def resolve_box_tags(box_obj, info):
+    return info.context["tags_for_box_loader"].load(box_obj.id)
 
 
 @query.field("product")
+def resolve_product(obj, _, id):
+    product = Product.get_by_id(id)
+    authorize(permission="product:read", base_id=product.base_id)
+    return product
+
+
 @box.field("product")
 @unboxed_items_collection.field("product")
-def resolve_product(obj, _, id=None):
-    product = obj.product if id is None else Product.get_by_id(id)
```

Comment:
why is this check not done / needed anymore?

Reply:
I split the resolvers for product-by-id and product-for-box. Previously they were combined but only for the first case an id is passed as argument.